### DISCUSSION TITLE
Can't query portable text annotations with Graphql

### DISCUSSION THREAD

**Discussion by madeleineostoja** [sanity-io/sanity#4097] [Sanity Studio Support]
I'm trying to add annotations to my portable text fields for internal linking, as described [here](https://www.sanity.io/guides/portable-text-internal-and-external-links). However I'm using graphQL to query Sanity, and the `{fieldname}Raw` fields don't return any data in the `markDefs`, just the type and key, eg: 

```
[{
  _key : "4c6d2d677368"
  _type: "link"
}]
```

Am I missing something? Is it not possible to use annotations properly with graphQL? And since the portabletext field is just a JSON blob there's no way to drill into it to resolve relations anyway correct?

**Comment by madeleineostoja** [comment]
Bump?

**Comment by madeleineostoja** [comment]
I love using sanity but man support options are pretty bad

**Comment by felixhagspiel** [comment]
if you do not want to use groq or gatsby, apparently you cannot use references in portable text. except if you want to do something like this: https://www.sanity.io/answers/handling-references-in-portable-text-in-graphql-without-gatsby-source-plugin
that is really bad, I now have to rewrite all my queries in groq, which is a very unintentional language compared to graphql. plus the fact that you do not get autocompletion on the playground makes sanity a pretty bad developer experience. which is a shame, because the I really like all the other things around sanity.
